pkg/client/package_buildpack.go

```diff
@@ -90,47 +125,88 @@ func (c *Client) PackageBuildpack(ctx context.Context, opts PackageBuildpackOpti
 
 	bpURI := opts.Config.Buildpack.URI
 	if bpURI == "" {
-		return errors.New("buildpack URI must be provided")
+		return digest, errors.New("buildpack URI must be provided")
+	}
+
+	if ok, platformRootFolder := buildpack.PlatformRootFolder(bpURI, target); ok {
+		bpURI = platformRootFolder
 	}
 
 	mainBlob, err := c.downloadBuildpackFromURI(ctx, bpURI, opts.RelativeBaseDir)
 	if err != nil {
-		return err
+		return digest, err
 	}
 
 	bp, err := buildpack.FromBuildpackRootBlob(mainBlob, writerFactory, c.logger)
 	if err != nil {
-		return errors.Wrapf(err, "creating buildpack from %s", style.Symbol(bpURI))
+		return digest, errors.Wrapf(err, "creating buildpack from %s", style.Symbol(bpURI))
 	}
 
 	packageBuilder.SetBuildpack(bp)
 
+	platform := target.ValuesAsPlatform()
+
 	for _, dep := range opts.Config.Dependencies {
+		if multiArch {
+			locatorType, err := buildpack.GetLocatorType(dep.URI, opts.RelativeBaseDir, []dist.ModuleInfo{})
+			if err != nil {
+				return digest, err
+			}
+			if locatorType == buildpack.URILocator {
+				// When building a composite multi-platform buildpack all the dependencies must be pushed to a registry
+				return digest, errors.New(fmt.Sprintf("%s is not allowed when creating a composite multi-platform buildpack, push your dependencies to a registry and use 'docker://' instead", style.Symbol(dep.URI)))
+			}
+		}
+
+		c.logger.Debugf("Downloading buildpack dependency for platform %s", platform)
 		mainBP, deps, err := c.buildpackDownloader.Download(ctx, dep.URI, buildpack.DownloadOptions{
 			RegistryName:    opts.Registry,
 			RelativeBaseDir: opts.RelativeBaseDir,
-			ImageOS:         opts.Config.Platform.OS,
 			ImageName:       dep.ImageName,
 			Daemon:          !opts.Publish,
 			PullPolicy:      opts.PullPolicy,
+			Target:          &target,
 		})
-
 		if err != nil {
-			return errors.Wrapf(err, "packaging dependencies (uri=%s,image=%s)", style.Symbol(dep.URI), style.Symbol(dep.ImageName))
+			return digest, errors.Wrapf(err, "packaging dependencies (uri=%s,image=%s)", style.Symbol(dep.URI), style.Symbol(dep.ImageName))
 		}
 
 		packageBuilder.AddDependencies(mainBP, deps)
 	}
 
 	switch opts.Format {
 	case FormatFile:
-		return packageBuilder.SaveAsFile(opts.Name, opts.Config.Platform.OS, opts.Labels)
+		name := opts.Name
+		if multiArch {
```

Comment:
I was just trying multi-arch buildpack package with this branch (the full command I tried looks something like `pack buildpack package --target linux/amd64 --target linux/arm64 --format file --verbose --path buildpack.tgz buildpack.cnb`) and it failed with the error:
```
ERROR: '''' is not a valid manifest reference: could not parse reference:
```

I added a few debug prints and from what I gathered this is happening because the `digest` value is never set for `case FormatFile`.

Reply:
Oh.. thanks! I will take a look on that!

Reply:
I fixed! 

When exporting to disk we shouldn't try to create an Image Index, only creating the files according to the RFC